# Dataset Comparison for Real Estate Recommendation System & AI Agent

This report analyzes the four available datasets to determine which best supports building a **Recommendation System** and an **AI Agent Chatbot**.

### Datasets Analyzed:
1. `Cario_properties_pricing.csv`
2. `Egypt_house_pricing.csv`
3. `egypt_real_estate_listings.csv`
4. `propertyfinder.csv`

In [1]:
import pandas as pd

df_cairo = pd.read_csv('Cario_properties_pricing.csv')
df_egypt = pd.read_csv('Egypt_house_pricing.csv')
df_listings = pd.read_csv('egypt_real_estate_listings.csv')
df_pf = pd.read_csv('propertyfinder.csv')

print(f"Cairo Properties: {df_cairo.shape}")
print(f"Egypt House Pricing: {df_egypt.shape}")
print(f"Real Estate Listings: {df_listings.shape}")
print(f"Property Finder: {df_pf.shape}")

c:\Users\Adham\anaconda3\Lib\site-packages\pandas\core\computation\expressions.py:22: UserWarning: Pandas requires version '2.10.2' or newer of 'numexpr' (version '2.8.7' currently installed).
  from pandas.core.computation.check import NUMEXPR_INSTALLED
c:\Users\Adham\anaconda3\Lib\site-packages\pandas\core\arrays\masked.py:56: UserWarning: Pandas requires version '1.4.2' or newer of 'bottleneck' (version '1.3.7' currently installed).
  from pandas.core import (
C:\Users\Adham\AppData\Local\Temp\ipykernel_20932\2033067359.py:6: DtypeWarning: Columns (0: completion_status) have mixed types. Specify dtype option on import or set low_memory=False.
  df_pf = pd.read_csv('propertyfinder.csv')


Cairo Properties: (11418, 7)
Egypt House Pricing: (54329, 11)
Real Estate Listings: (19924, 11)
Property Finder: (39713, 53)


## 1. Cario Properties Pricing
**Shape:** ~11,418 rows $\times$ 7 columns  
**Columns:** `type`, `title`, `location`, `bedroom`, `bathroom`, `size_sqm`, `price`

**Analysis:**
- **Pros:** Simple, clean, quick to process.
- **Cons:** Lacks detailed text descriptions, amenities, and geographical coordinates. Too limited for a modern conversational AI agent or a content-based recommendation system.

## 2. Egypt House Pricing
**Shape:** ~54,329 rows $\times$ 11 columns  
**Columns:** `ad_id`, `type`, `price`, `area`, `bedrooms`, `bathrooms`, `level`, `furnished`, `rent`, `city`, `region`

**Analysis:**
- **Pros:** High volume of data; has granular categorical variables (`furnished`, `level`, `city`). Useful for collaborative filtering or basic strict-filter matching.
- **Cons:** Missing free-text descriptions entirely. A chatbot would sound very robotic if it only had these categorical variables to discuss properties with the user.

## 3. Egypt Real Estate Listings
**Shape:** ~19,924 rows $\times$ 11 columns  
**Columns:** `url`, `price`, `description`, `location`, `type`, `size`, `bedrooms`, `bathrooms`, `available_from`, `payment_method`, `down_payment`

**Analysis:**
- **Pros:** Contains `description`! This is critical for natural language processing (NLP). The chatbot can use Retrieval-Augmented Generation (RAG) to understand property vibes. Includes `payment_method` which is highly relevant to users.
- **Cons:** The feature space is still somewhat small compared to an industry-standard real estate API.

## 4. Propertyfinder
**Shape:** ~39,713 rows $\times$ 53 columns  
**Columns:** `listing_id`, `detail_url`, `property_type`, `price_egp`, `location_full`, `lat`, `lon`, `bedrooms`, `bathrooms`, `area_value`, `furnished`, `images_count`, `description`, `amenities`, `payment_method`, `agent_name`, `broker_name`, etc.

**Analysis:**
- **Pros:** 
    - **Extensive Metadata:** 53 columns. It covers absolutely everything from precise coordinates (`lat`/`lon`) to `amenities`, `description`, and `images_count`.
    - **Rich Context for AI:** An AI chatbot can perform advanced querying ("Show me apartments in Maadi with a balcony and 3 bedrooms under 5M EGP") and provide rich, conversational answers using the `description` and `amenities`.
    - **Better Recommendations:** You can build hybrid recommendation systems (Collaborative Filtering using listings + Content-Based Filtering using TF-IDF/Sentence Transformers on `amenities` and `description`).
- **Cons:** Data cleaning will require more effort (missing values, parsing lists from `amenities` column).

## Final Recommendation

### 🏆 Winner: **`propertyfinder.csv`**

For both a **Recommendation System** and an **AI Chatbot Agent**, you must use **`propertyfinder.csv`**.

**Why for the Recommendation Engine?**
We have access to `amenities`, `lat`, `lon`, and `property_type`. You can cluster properties geographically using lat/lon and calculate similarity scores based on NLP embeddings of the text data.

**Why for the AI Agent?**
A Large Language Model Agent needs context to sound intelligent. Having `agent_name`, `furnished` status, and raw rich `description` text gives the language model exactly what it needs to perform natural, conversational RAG (Retrieval Augmented Generation).

In [2]:
# Let's explore the recommended dataset
df_pf.head()

,listing_id,internal_id,category,listing_type,detail_url,property_type,offering_type,completion_status,title,price_egp,...,agent_is_super,agent_languages,broker_id,broker_name,broker_email,broker_phone,contact_phone,contact_whatsapp,contact_email,scraped_at
0,F7QB31CGWE509V2W7DF2GARB2C,56009081.0,buy,property,https://www.propertyfinder.eg/en/plp/buy/duple...,Duplex,Residential for Sale,completed,Garden Villa - Lake View Boutique - Prime Loca...,24500000.0,...,False,NaN,5758.0,Spade consultancy,mohmedg.sedik@gmail.com,1.001437e+09,2.012018e+11,2.022126e+10,pierre.Osama@spade-consultancy.com,2026-03-04T14:20:33.281007
1,K1JC3D6N57ED52N3VX1QQKHHXG,56247925.0,buy,property,https://www.propertyfinder.eg/en/plp/buy/apart...,Apartment,Residential for Sale,off_plan,For Sale: Finished Apartment+ ACs in Village West,5145000.0,...,False,NaN,492.0,Abrag Real Estate,snawara71@gmail.com,2.010012e+11,2.010072e+11,2.022126e+10,ranoushamer901@gmail.com,2026-03-04T14:20:33.281007
2,Q6GEB8T6PZTJGNNPWA5PX3JCWR,56253883.0,buy,property,https://www.propertyfinder.eg/en/plp/buy/apart...,Apartment,Residential for Sale,completed,UnderMarket Price for Apt. 250 RTM PrimeLocation,10800000.0,...,False,English | Arabic,5279.0,Premises,mahmouddessam@hotmail.com,2.010051e+11,2.010910e+11,2.022126e+10,ashrafsobhy99@hotmail.com,2026-03-04T14:20:33.281007
3,PPH0852SAB266X6GSVATJP4F0M,56250852.0,buy,property,https://www.propertyfinder.eg/en/plp/buy/apart...,Apartment,Residential for Sale,completed,lowest price 2bds fullyfurnished @ lake view,10500000.0,...,False,NaN,5758.0,Spade consultancy,mohmedg.sedik@gmail.com,1.001437e+09,2.010701e+11,2.022126e+10,ebrahim.hossam@spade-consultancy.com,2026-03-04T14:20:33.281007
4,QCSA4TNDEBBV7BNY8ANST3TB50,55995764.0,buy,property,https://www.propertyfinder.eg/en/plp/buy/villa...,Villa,Residential for Sale,completed,Exclusive Type M with Lowest Price RTM in PNC,19500000.0,...,False,English | Arabic,5279.0,Premises,mahmouddessam@hotmail.com,2.010051e+11,2.010910e+11,2.022126e+10,ashrafsobhy99@hotmail.com,2026-03-04T14:20:33.281007


In [3]:
pd.set_option('display.max_columns', None)
df_pf[['property_type', 'price_egp', 'location_full', 'amenities', 'description']].head(3)

,property_type,price_egp,location_full,amenities,description
0,Duplex,24500000.0,"The Lakeview Boutique Villas, 5th Settlement C...",Balcony | Built in Wardrobes | Covered Parking...,Garden Villa for Sale – Featured in Lake View ...
1,Apartment,5145000.0,"Village West, Sheikh Zayed Compounds, Sheikh Z...",Balcony | Built in Wardrobes | Central A/C | C...,For Sale: Finished Apartment with ACs in Villa...
2,Apartment,10800000.0,"Palm Hills New Cairo, 5th Settlement Compounds...",Balcony | Built in Wardrobes | Central A/C | C...,Palm Hills New Cairo Apartment BUA: 250sqm 4 B...
